src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -27,11 +27,19 @@ Copyright (c) .NET Foundation. All rights reserved.
        by default since they may require assemblies, types or members that
        could be removed by the linker, causing a trimmed app to crash. -->
   <PropertyGroup Condition="'$(StartupHookSupport)' == '' And
-                            '$(PublishTrimmed)' == 'true' And
```

Comment:
@eerhardt FYI I moved this check into the RuntimeConfigurationOptions since I found that if you publish an app without /p:PublishTrimmed=true, but the project template defaults to publishTrimmed (like blazorwasm) then there is a race condition that causes PublishTrimmed to be set to true too late by the SDK, causing this condition to always be false. By moving it down to the RuntimeConfigurationOptions instead, I have validated that the condition evaluates succesfully at that point and does get the right behavior for blazorwasm projects as well.

Reply:
FYI: @mateoatr

Reply:
We will need to find a different solution for this (maybe change Blazor to set the values earlier) - if we use the property for anything later on, we can't rely on its value being correct - since this will effectively set it to false for all .NET 6 apps. It will create lot of confusion. Also see the comment about setting the runtime option even when not trimming.

Reply:
Yeah, I have spoken with @pranavkm and the solution will be to moved PublishTrimmed setting for wasm apps  up to the props files instead of targets. I'm adding that change as part of this PR.